### DISCUSSION TITLE
Server optimisation for best performance - core set of recommendations?

### DISCUSSION THREAD

**Discussion by danielrosehill** [nocodb/nocodb#8988] [General]
Hi everyone,

Wondering whether there is any documentation available (or that perhaps could be formulated by the community) describing both a minimal and optimal set of server requirements for provisioning a self-hosted NoCoDB instance with a locally hosted database (ie, NoCoDB + Postgres deployment).

Would be interested in knowing both an MVP-scale and production-scale description for:

- OS recommendation (distro and variant)
- RAM
- Storage

Additionally, would be good to know what the consensus is around the best way to deploy a stable production instance as hosting both the database and app as Docker images seems (to me at least) not ideal (but especially the database).

TIA for any suggestions!

**Comment by danielrosehill** [comment]
This is what I'm currently running on. Feel like that should be more than enough although it still feels a little sluggish, at least relative to Airtable. But perhaps the issue isn't infra-related:

![image](https://github.com/user-attachments/assets/a1db540a-4177-464b-94da-247779dc51f5)

**Reply by o1lab** [reply]
Hey @danielrosehill, nocodb is very light - many people even run it on raspberry pi and sqlite :)

the above config is more than sufficient for <50 users. And hetzner vCPU is atleast 1.5x of aws vCPU from our experiments. Only if you are going to use API really heavily you will need more CPU.

And here is the most easiest way to setup nocodb for production usage using docker compose - we will soon make it part of our docs.

`bash <(curl -sSL http://install.nocodb.com/noco.sh) <(mktemp)`

This automatically 
- Installs all pre-requisites like docker, docker-compose, nginx on any linux distro.
- Installs nocodb with SSL (you will have to provide domain or subdomain name)
- Auto upgrades when new releases happpen
- Auto renews SSL
- Has helper script to upscale/downscale nocodb, show logs etc

And from our experiments, using a native postgres yields better performance than docker volume mounted pg (as one can expect)

**Comment by danielrosehill** [comment]
@o1lab / Navi,

First and foremost and above all: thank you for building a truly outstanding tool. I'll likely chip in with a few feature suggestions and try to support the project however I can over the coming weeks.

Secondly .... thanks for clarifying re the specs. That makes sense. I expect 5 or 6 users at best. 

Playing around with things a bit more ... I think that the lagginess I'm experiencing is definitely due to the database. 

Any chance you have any tips/resources for migrating seamlessly from a docker Postgres over to an on-disk/native one?

I think I'll get that out of the way early in the deployment before it becomes a bigger headache later!

Many thanks for the quickly reply!

Daniel

**Reply by o1lab** [reply]
pg_dump will easily do

**Reply by o1lab** [reply]
And thank you for words of encouragement - will pass it on to the team.

What is the size of teh database or tables ? are you applying lot of filters?
